### PR TITLE
saga, pdal: update saga to version 9.0.2 and pdal to 2.5.4

### PR DESCRIPTION
**Pull request by nilason**
#### Description

Transition to CMake build system.

Closes https://trac.macports.org/ticket/62920

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.5 21G531 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by nilason** [comment]
I add to this PR an update of PDAL.

File: gis/saga/Portfile [hunk 1]
```diff
@@ -1,126 +1,206 @@
...
+                    port:proj9 \
+                    port:opencv4 \
+                    port:PDAL \
+                    port:curl \
+                    port:tiff
+
+mpi.setup
+
+compiler.cxx_standard       2017
+configure.cxxflags-append   -std=c++17
```
Comment (reneeotten) [inline gis/saga/Portfile:52]: does the cmake PG not take care of this - I think it's supposed to

File: gis/saga/Portfile [hunk 2]
```diff
@@ -1,126 +1,206 @@
...
+                            -DWITH_SYSTEM_GEOTRANS=OFF \
+                            -DWITH_CLIPPER_ONE=OFF \
+                            -DWITH_MRMR=OFF
+                            
+configure.args-append       -DWITH_TOOLS=ON \
+                            -DWITH_TOOLS_OPENCV=ON \
+                            -DWITH_TOOLS_GDAL=ON \
+                            -DWITH_TOOLS_PDAL=OFF \
+                            -DWITH_TOOLS_HPDF=OFF \
+                            -DWITH_TOOLS_PDAL=ON \
```
Comment (reneeotten) [inline gis/saga/Portfile:74]: duplicated line, above it's set to OFF

File: gis/saga/Portfile [hunk 3]
```diff
@@ -1,126 +1,206 @@
...
+mpi.setup
+
+compiler.cxx_standard       2017
+configure.cxxflags-append   -std=c++17
+
+configure.pkg_config_path-append \
+                            "${prefix}/lib/opencv4/pkgconfig"
+
+configure.cppflags-append   -I${prefix}/lib/proj9/include
+configure.ldflags-append    -L${prefix}/lib/proj9/lib
```
Comment (reneeotten) [inline gis/saga/Portfile:58]: same here for adding to configure flags; this was in the original Portfile but does CMake bot figure this out on its own?

**Review by reneeotten — COMMENTED** [review]
I haven't tested anything locally but just a few comments after a quick read through

File: gis/saga/Portfile (hunk 1 above)
Comment (nilason) [reply gis/saga/Portfile]: It does, and SAGA actually requires c++14. Changed accordingly.

File: gis/saga/Portfile (hunk 2 above)
Comment (nilason) [reply gis/saga/Portfile:72]: Nice catch!

File: gis/saga/Portfile (hunk 3 above)
Comment (nilason) [reply gis/saga/Portfile]: Good you noticed this! Turned out SAGA only support PROJ pre-8 API. I fixed the version used to proj7 and added cmake flags (which are needed as PROJ files are in non-standard location).

**Comment by reneeotten** [comment]
@Veence I'll merge this on Wednesday unless you comment beforehand.

**Comment by nilason** [comment]
> @Veence I'll merge this on Wednesday unless you comment beforehand.

Please, merge if there are no other issues.

**Comment by reneeotten** [comment]
sorry @nilason - keeping track of days is difficult 😉

**Comment by nilason** [comment]
> sorry @nilason - keeping track of days is difficult 😉

No worries, thanks a bunch!

**Pull request merged**